File: faas-fn/countv3/src/main/java/com/openfaas/function/Handler.java
package com.openfaas.function;
import io.vertx.ext.web.RoutingContext;
import io.vertx.core.json.JsonObject;

public class Handler implements io.vertx.core.Handler<RoutingContext> {

  public void handle(RoutingContext routingContext) {
     // Use bodyHandler to handle the request body
        routingContext.request().bodyHandler(buffer -> {
            try {
                // Convert the buffer to a String
                // String requestBody = buffer.toString();

                // // Log the received JSON
                // System.out.println("Received JSON: " + requestBody);

                // // Create a JsonObject from the request body
                // JsonObject jsonResponse = new JsonObject(requestBody);
                // if (jsonResponse.containsKey("number")) {
                //     // Extract the "number" field from the JSON payload
                //     int inputNumber = jsonResponse.getInteger("number");

                //     // Increment the number by 1
                //     int result = inputNumber + 1;

                //     // Prepare the response JSON
                //     jsonResponse = new JsonObject()
                //             .put("status", "ok")
                //             .put("result", result);
                JsonObject jsonResponse = new JsonObject().put("status", "ok").put("result", "How you doing ? ");
                
                // Set the response content type and send the response
                routingContext.response()
                        .putHeader("content-type", "application/json")
                        .putHeader("Access-Control-Allow-Origin", "*")
                        .putHeader("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
                        .putHeader("Access-Control-Allow-Headers", "Content-Type, Authorization")
                        .end(jsonResponse.encodePrettily());
            } catch (Exception e) {
                // Handle any exceptions
                System.err.println("Error processing request: " + e.getMessage());
                routingContext.response()
                        .setStatusCode(500) // Internal Server Error
                        .putHeader("content-type", "application/json;charset=UTF-8")
                        .end(new JsonObject().put("error", "it gets in the function but then fails").encodePrettily());
            }
        });
    }
}

File: faas-fn/countv4/handler.py
import json


def handle(event, context):
    try:
        request_body = json.loads(event.body)


        if 'number' in request_body:
            # Extract the "number" field from the JSON payload
            input_number = request_body['number']

            # Increment the number by 1
            result = input_number + 1

            # Prepare the response JSON
        response = {
            'statusCode': 200,
            'headers': {
                'content-type': 'application/json',
                'Access-Control-Allow-Origin': '*',
                "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
                "Access-Control-Allow-Headers": "Content-Type, Authorization",
                "Access-Control-Allow-Credentials": "true"
            },
            'body': json.dumps({
                'status': 'ok',
                'result': result
            })
        }
        return response
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'content-type': 'application/json',
                'Access-Control-Allow-Origin': '*',  # Replace with your allowed origin
                'Access-Control-Allow-Methods': 'GET, POST, OPTIONS',
                'Access-Control-Allow-Headers': 'Content-Type, Authorization',
                'Access-Control-Allow-Credentials': 'true'
            },
            'body': json.dumps({
                'error': 'Internal Server Error'
            })
        }
